### PR TITLE
Bug Fix: Empty combo box on Component change

### PR DESCRIPTION
**Pull request by amzn-ahmadkrm**
## What does this PR do?

The custom widgets `CollisionLayerWidget`, `CollisionGroupWidget` and `MeshNodeHandler` were recently changed to indirectly inherit from `GenericComboBoxCtrl`. This created a bug where the widgets were trying to clear and re-add elements to the ComboBox but the values were not being re-added because they weren't being cleared from the values stored in the `GenericComboBoxCtrl`.

_Please describe your PR. For a bug fix, what was the old behavior, what is the new behavior?_

When the component changed the fields were cleared and not repopulated, this PR corrects the behavior of the elements being removed so that it can work as intended.

Editor_HxfCxOfiT9

## How was this PR tested?

Manual testing in editor.

### CODE REVIEW

**Review by amzn-tmryan — APPROVED** [review]

**Review by dtamkin1 — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/GenericComboBoxCtrl.h [hunk 1]
```diff
@@ -92,6 +92,7 @@ namespace AzToolsFramework
         void addElements(const AZStd::vector<AZStd::pair<T, AZStd::string>>& genericValues);
         void addElement(const AZStd::pair<T, AZStd::string>& genericValue);
         void setElements(const AZStd::vector<AZStd::pair<T, AZStd::string>>& genericValues);
+        void clearElements();
```
Comment (cgalvan) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/GenericComboBoxCtrl.h:95]: Nit: We prefer upper camel-case for our methods.
```suggestion
        void ClearElements();
```

**Review by cgalvan — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/GenericComboBoxCtrl.h (hunk 1 above)
Comment (burelc-amzn) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/GenericComboBoxCtrl.h:95]: All the rest of the methods in this class use the Qt-style `lowerCamelCase`, I'd think the rule to follow the style of the rest of the existing code would apply here.

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyStringComboBoxCtrl.cpp [hunk 2]
```diff
@@ -46,7 +46,7 @@ namespace AzToolsFramework
 
     void PropertyStringComboBoxCtrl::Add(const AZStd::vector<AZStd::string>& vals)
     {
-        GetComboBox()->clear();
+        ComboBoxBase::clearElements();
```
Comment (burelc-amzn) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyStringComboBoxCtrl.cpp:49]: Why do you need to qualify this call with `ComboBoxBase::`? I'd expect this to be just
```suggestion
        clearElements();
```

**Review by moraaar — APPROVED** [review]
LGTM

**Comment by hultonha** [comment]
Temporarily moving to Draft while the author is away - will merge on their return

**Pull request merged**